core/src/engine/task/task_processor.cpp

```diff
@@ -68,11 +70,15 @@ void TaskProcessorThreadStartedHook() {
 TaskProcessor::TaskProcessor(TaskProcessorConfig config,
                              std::shared_ptr<impl::TaskProcessorPools> pools)
     : task_counter_(config.worker_threads),
-      task_queue_(config),
+      task_queue_(std::in_place_index_t<1>(), config),
```

Comment:
Don't create old queue, then destroy and create new queue if config

Create the correct queue right away

```cpp
namespace {

auto MakeTaskQueue(TaskProcessorConfig&& config) {
  using ResultType = std::variant<...>;
  switch (...) {  // Review note: please use switch when working with enums
    case ...:
      return ResultType{std::in_place_index<0>, std::move(config)};
    ...
  }
  UINVARIANT(false, "Unexpected value of ... enum");
}

}